From 4ddde78b86adcea54c73979d3ba0b16821c5aedc Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sun, 9 Jun 2019 17:36:22 +0200
Subject: [PATCH] gluon-mesh-vpn-core: don't consider tunneldigger when fastd
 is installed

None of our scripts can handle fastd and tunneldigger running at the same
time. When both are installed, ignore tunneldigger.

(cherry picked from commit f6df3b0541338461fc40898e3570f49cbea0bf22)
---
 .../gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn b/package/gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn
index 00764588..0933d67b 100755
--- a/package/gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn
+++ b/package/gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn
@@ -52,7 +52,7 @@ uci:save('firewall')
 local has_fastd = unistd.access('/lib/gluon/mesh-vpn/fastd')
 local fastd_enabled = uci:get('fastd', 'mesh_vpn', 'enabled')
 
-local has_tunneldigger = unistd.access('/lib/gluon/mesh-vpn/tunneldigger')
+local has_tunneldigger = (not has_fastd) and unistd.access('/lib/gluon/mesh-vpn/tunneldigger')
 local tunneldigger_enabled = uci:get('tunneldigger', 'mesh_vpn', 'enabled')
 
 local enabled
-- 
2.17.1

